core/commands/generate_command.rb

```diff
@@ -447,7 +446,16 @@ def generate_vagrant_file(path, config, boxes, provider, cookbook_path)
       end
     end
     vagrant.puts vagrant_config_footer
+  rescue RuntimeError => e
+    @ui.error(e.message)
+    @ui.error('Configuration is invalid')
+    @env.aws_service.delete_key_pair(keypair_name) if provider == 'aws'
     vagrant.close
+    FileUtils.rm_rf(path)
+    ERROR_RESULT
+  else
+    vagrant.close
```

Comment:
This code should be moved up, so it would be before the rescue clause. I see no point in using else in this context